sanchay811/Pander
# Firebase Setup

- Removed mock server files
- Created reusable routes and constants files
- Expanded on reusable Text-Input component
- Introduced FireBase integration, simple sign up setup
- Added additional dummy route points

**Steps to test SignUp w/ Firebase**
(Install dependencies)
1. Go to https://firebase.google.com/ and create a free account, you will have to setup a sample project to connect to this app. 
2. Then once a project is setup click on Authentication under Develop in the side navbar on the firebase-console page. Under Sign-in method you want to enable email/password.
3. After checking out this branch, create your own .env file in the following format:
```javascript
API_KEY="YOUR_KEY"
AUTH_DOMAIN="YOUR_AUTH_DOMAIN"
DATABASE_URL="YOUR_DATABASE_URL"
PROJECT_ID="YOUR_PROJECT_ID"
STORAGE_BUCKET="YOUR_STORAGE_BUCKET"
MESSAGE_SENDER_ID="YOUR_MESSAGE_SENDER_ID"
```
4.  Update your firebase config values:
Your config information can be found once logged on the firebase website -> have a sample project setup -> in the firebase console -> navigate to project overview and click the settings icon (in side-nav) -> click on project settings -> scroll down and click on the icon with opening and closing tag, you will see your config information here. 
Copy this info to your .env file. File Location: rootFolder/.env

5. Start the app
6. Navigate to end-point localhost:8080/sign-up
7. Fill in details and submit form
8. Check your firebase console under Develop - Authentication, you should see the information submitted from the form
